webapp/lib/view.dart

```diff
@@ -575,10 +574,45 @@ class ConversationListSelectHeader {
         ..text = 'Conversation List:');
 
     panel.append(
-      new SelectElement()
+      _selectElement = new SelectElement()
         ..classes.add('conversation-list-select')
-        // TODO dynamically populate this from firebase "tables/conversation-lists"
-        ..add(OptionElement(data: 'nook_conversations'), null));
+        ..add(OptionElement(data: '... loading conversation lists ...'), null)
+        ..onChange.listen(shardSelected));
+  }
+
+  void updateConversationLists(List<ConversationListShard> shards) {
+    // TODO Consider displaying summary information about each shard... e.g. # of unread conversations
+    if (_shards == null) {
```

Comment:
Doesn't this mean that if the list of shards changes in firebase, it won't update in the UI without a page refresh?

Reply:
I think ideally we would like it to update in the conversation list selector automatically, and if possible, to keep the user on the shard they're on etc. but this is probably something to follow up in a future PR

Reply:
@marianamarasoiu There are two ways that the shards can change

* name or status information in the shard description changes in which case, yes, we should refresh the conversation list

* a new shard is added which means that conversations have moved to different shards. In this situation we should probably force refresh the page so that it displays reality again. We need to guard against a client with out of date shards unknowingly sending invalid changes to the backend

I am not clear or sure about the above so I left it for a future PR.